packages/auth/src/accessToken/service.ts

```diff
@@ -159,10 +164,11 @@ async function createAccessToken(
 
 async function rotate(
   deps: ServiceDependencies,
-  managementId: string
+  managementId: string,
+  tokenValue: string
 ): Promise<Rotation> {
   let token = await AccessToken.query(deps.knex).findOne({ managementId })
-  if (token) {
+  if (token && token.value === tokenValue) {
     await token.$query(deps.knex).delete()
     token = await AccessToken.query(deps.knex).insertAndFetch({
```

Comment:
I think the `findOne` and `delete` should otherwise use a lock.
```
  const oldToken = await AccessToken.query(deps.knex)
    .delete()
    .returning('*')
    .findOne({
      managementId,
      value: tokenValue
    })
  if (oldToken) {
    const token = await AccessToken.query(deps.knex).insertAndFetch({
```
The order of the `delete`/`returning`/`findOne` is a workaround for:
- https://github.com/Vincit/objection.js/issues/1649#issuecomment-575698815

Reply:
:thinking: 
Maybe this should all happen inside a transaction though.
Both the delete and insert.